### ISSUE TITLE
Docs: [no-floating-promises] voiding Promise doesn't make it handled

### ISSUE THREAD

**Issue by alexandercerutti** [typescript-eslint/typescript-eslint#9947]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Hi!

[[no-floating-promises]](https://typescript-eslint.io/rules/no-floating-promises/) cite what follows:

```
[...]

This rule reports when a Promise is created and not properly handled. Valid ways of handling a Promise-valued statement include:

[...]
- voiding it
[...]
```

However, **voiding a Promise doesn't actually make it handled**. The result is just ignored.

If a voided Promise is rejected, an error is still thrown and thus the failure is **just ignored**. However, the failure is still there. In fact:
 - a listener for `unhandledrejection` still gets invoked;
 - an error is still logged in the console and eventually in enterprise logging systems (e.g. Sentry);

Therefore, it is not exactly correct to say that `void` operator is a valid way of handling a Promise. **A promise should still be followed by a .catch() (even with a noop function) right after the invokation (not after or in an if)**.

This code proves what happens. I trusted this package, I voided some Promises I didn't expect could reject and, after enabling Sentry, I ended up with a ton of reports about unhandled rejections.

```javascript
	window.addEventListener("unhandledrejection", (err) => {
		console.log("Unhandled Rejection", err);
	});

	async function f1() {
		return Promise.reject("Timeout expired");
	}

	void f1();
	console.log("test");
```

Logs orders:
 - (log) `"test"`
 - (log) `"Unhandled Rejection"`
 - (error) `Uncaught (in promise) Timeout expired`

<img width="844" alt="immagine" src="https://github.com/user-attachments/assets/ab8acfda-46fb-4b1a-943f-db1e95515795">

I'd improve the documentation by putting a warning on void about this behavior, both above in the list (add a link to `ignoreVoid`) and under the `ignoreVoid`.

Runtime error messages might get improved as well, but as I'm on an older version of the extension (v6 if I see correctly), I still went to see the documentation to check if the usage is correct.

Code could be improved by requiring the `.catch(() => { ... })` even for voided Promises or by setting `{ ignoreVoid: false }` by default, as the behavior is potentially annoying.

Thank you

### Affected URL(s)

https://typescript-eslint.io/rules/no-floating-promises/

**Comment by Josh-Cena** [comment]
What we mean is "valid ways to silence this rule". We aren't really interested in going into details about how you should handle errors and what they do exactly.

**Comment by alexandercerutti** [comment]
@Josh-Cena first of all, thank you for your reply.

> What we mean is "valid ways to silence this rule"

That's a completely different thing. Silencing doesn't imply handling in this case, if you include `"void"` and `.then()` with one argument (docs says two, so it's fine).

I'm going to open a PR to fix this.

**Comment by Josh-Cena** [comment]
> That's a completely different thing.

Yeah that's what I meant, the current one is wrong and we should say "the ways to silence the linter" because we aren't teaching you how to handle errors here.
